Esri/arcgis-maps-sdk-swift-samples
# [New] Snap geometry edits

## Description

This PR implements `Snap geometry edits` in `Edit and Manage Data` category.

URL to README: https://github.com/Esri/arcgis-maps-sdk-swift-samples/blob/Ting/New-Snapping/Shared/Samples/Snap%20geometry%20edits/README.md

## Linked Issue(s)

- `swift/issues/5084`
- Kotlin https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/194

## How To Test

- Open the sample, use the geometry editor to draw all kinds of geometries
- After seeing the normal behavior, turn on snapping for all layers
- Continue drawing geometries
    - On iOS, when you drag a vertex to a feature, it should snap with highlight;
    - On Mac Catalyst, hover the mouse and it snaps
- Turn off some layers, and see they don't snap anymore.

## Screenshots

![screenshot](https://raw.githubusercontent.com/Esri/arcgis-maps-sdk-swift-samples/dd49ce1f44df3a699d2f78e2b7c124bf18c315c9/Shared/Samples/Snap%20geometry%20edits/snap-geometry-edits.png)

## To Discuss

- We cannot reuse types from other samples, so unfortunately while https://github.com/Esri/arcgis-maps-sdk-swift-samples/pull/372/commits/db920126846151efdac5229b8fc6f60aa3cb2dfd is short and concise, I have to copy everything from `Create and edit geometries` to use the geometry editor. 😢 
- All the geometry editor code are copy pasted from the sample with only renaming and doc changes.
- "main", "lateral" are the thick and thin azure polylines; "system valve" are the green and red cat eyes; "service connection" are the "M"s; "curb stop valves" are the "x" shaped circles (need to zoom in to curb scale).
- I didn't include the "Tap to edit graphic" feature from the original design as I think it is not relevant to this sample.
